ereolen_support/ereolen_support.helper.inc

```diff
@@ -0,0 +1,255 @@
+      // Look for form fields.
+      if(strpos($key, 'ereolen_support') === 0) {
+        // Load config related to that field.
+        $jira_field_id = variable_get($key . '_jira_field');
+        // Make sure the config is set and has proper type.
+        if (!empty($jira_field_id) && is_numeric($jira_field_id)) {
+          // Add custom field to data array.
+          $data['fields']['customfield_' . $jira_field_id] = $form_state['values'][$key];
+          if($element['#type'] == 'select') {
+            // Use Label as output from select lists.
+            $data['fields']['customfield_' . $jira_field_id] = $element['#options'][$form_state['values'][$key]];
+          }
+          else {
+            $data['fields']['customfield_' . $jira_field_id] = $form_state['values'][$key];
+          }
+        }
+      }
+    }
+  }
+
+  $requestParts = _ereolen_support_get_request_parts('POST', $data);
+
+  $result = drupal_http_request($requestParts['jira_path'] . '/issue', $requestParts['options']);
+  _ereolen_support_log_error($result);
+}
+
+/**
+ * Create the parts required to make a drupal_http_request.
+ *
+ * @param string $method_type
+ *   The request method.
+ * @param array $data
+ *   An optional data array.
+ *
+ * @return array
+ *   An array containing base path for the request and the options related to
+ *   the request.
+ *
+ * @return array
+ */
+function _ereolen_support_get_request_parts($method_type, $data = []) {
+  $jiraUsername = variable_get('jira_username');
+  $jiraPassword = variable_get('jira_password');
+  // Jira requires base64_encode auth.
+  $encoded_auth = base64_encode($jiraUsername . ':' . $jiraPassword);
+  $jira_api_path = variable_get('jira_domain') . '/rest/api/latest';
+
+  $options = array(
+    'method' => $method_type,
+    'timeout' => 15,
+    'headers' => array(
+      'Content-Type' => 'application/json',
+      'Authorization' => 'Basic '. $encoded_auth
+    ),
+  );
+
+  // Add data variable to POST and PUT requests.
+  if ($method_type == 'POST' || $method_type == 'PUT' ) {
+    $options['data'] = json_encode($data);
+  }
+
+  $requestParts = array(
+    'jira_path' => $jira_api_path,
+    'options' => $options
+  );
+
+  return $requestParts;
+}
+
+/**
+ * Prepare an array for use in a select list.
+ *
+ * @param string $vocab_machine_name
+ *  A vocabulary machine name.
+ * @return array
+ *   A list of term names.
+ */
+function _ereolen_support_create_select_list($vocab_machine_name) {
+  $vacabulary = taxonomy_vocabulary_machine_name_load($vocab_machine_name);
+  $terms = taxonomy_get_tree($vacabulary->vid);
+  if ($terms) {
+    $list = [];
+    foreach ($terms as $term) {
+      $list[$term->tid] = str_repeat('-', $term->depth) . $term->name;
+    }
+
+    return $list;
+  }
+  return [];
+}
+
+/**
+ * Provide a support ID grabbed from Drupal user.
+ *
+ * @param \stdClass $user
+ *   A drupal user.
+ * @return null
+ */
+function _ereolen_support_get_support_id($user) {
```

Comment:
Maybe this should be `_ereolen_support_get_user_support_id`?

Reply:
@todo ?

Reply:
Removed